File: Graph Data Structures/Python/BFS Traversal.py
def push(data, que):
    global start
    global end
    if start == end:
        start += 1
    end += 1
    que.append(data)

N = int(input("Enter No of Vertices in Graph : "))

graph = []
que = []
start = -1
end = -1
flag = [0]

print("Enter value of edges in matrix")
for i in range(N+1):
    arr = []
    if i == 0:
        for j in range(N+1):
            arr.append(0)
    else:
        for j in range(N+1):
            if j == 0:
                arr.append(0)
            else:
                print("Enter value in ", i, "th row and",j,"th column : ", end="")
                arr.append(int(input()))
    graph.append(arr)

for i in range(N+1):
    flag.append(-1)

startVertex = int(input("Enter the vertex to start from : "))

push(startVertex, que)
flag[startVertex] += 1

ptr = 0

while ptr != N:
    startVertex = que[ptr]
    print(startVertex, " ", end="")

    for i in range(N+1):
        if graph[startVertex][i] > 0 and flag[i] == -1:
            push(i, que)
            flag[i] += 1

    ptr += 1

File: Graph Data Structures/C++/BFS Traversal.cpp
#include <iostream>

using namespace std;

void push( int x, int qp[], int &start, int &end );

int main()
{
    int N;

    cout << "Enter No of Vertices in Graph : ";
    cin >> N;

    int graph[N+1][N+1];
    int q[N];
    int start = -1;
    int end = -1;
    int flag[N+1];
    int x;
    int ptr;

    cout << endl << "Enter Value of edges in Matrix" << endl;
    for( int i = 1; i <= N; i++ )
    {
        for( int j = 1; j <= N; j++ )
        {
            cin >> graph[i][j];
        }
    }

    for( int i = 1; i <= N; i++ )
    {
        flag[i] = -1;
    }

    cout << endl << "Enter the vertex to start from : ";
    cin >> x;

    push( x, q, start, end );
    flag[x]++;
    ptr = 0;

    while( ptr != N )
    {
        cout << x << " ";

        for( int i = 1; i <= N; i++ )
        {
            if( graph[x][i] > 0 && flag[i] == -1 )
            {
                push( i, q, start, end );
                flag[i]++;
            }
        }
        ptr++;
        x = q[ptr];
    }

    return 0;
}

void push( int x, int q[], int &start, int &end )
{
    if( start == end )
    {
        start++;
    }
    end++;
    q[end] = x;
}
